### DISCUSSION TITLE
Create actor directly using pv.PolyData() with some properties like color and opcaity

### DISCUSSION THREAD

**Discussion by zhengliuX** [pyvista/pyvista#1924] [Ideas]
Hi, I wonder if there is a method that we could create a actor directly using data like pv.PolyData(), then we could add the actor to the plotter when we want. This might be useful in GUI interface.

**Comment by banesullivan** [comment]
This isn't currently supported, and I would be somewhat tough/critical on any proposed changes to incorporate this functionality in PyVista.

With that said, there are some ways to work around this. Usually, the go to is to use `add_mesh()` with your poly data to add it to a plotter and create the actor, then set the visibility to 0.

```py
actor = p.add_mesh(poly, **display_params)
actor.SetVisibility(False)

...

# Now do whatever you want with the actor
```

----

In order to support this, we would like have to heavily refactor the rather complex and delicate `add_mesh` and `add_volume` methods (changes to these methods need to be done with extreme care)
